File: server/models/Company.model.js
const mongoose = require("mongoose"); //kết nối DBDB
const validator = require("validator"); //Thư viện kiểm tra ràng buộc
const { ObjectId } = mongoose.Schema.Types; // lấy ID của một đối tượng trong bảng gán cho ObjectId
const CompanySchema = new mongoose.Schema({
  //định nghĩa table temp chưa lưu vào DB (gom dữ liệu thành 1 bảng)
  name: {
    type: String,
    required: true, // == Not Null (1 chiều từ User --> DB)
  },
  phone: {
    type: String,
    required: true,
    unique: true, // == kiểm tra không trùng = true, trùng = false
    /*kiểm tra đúng kiểu SDT*/
    validate: {
      validator: function (number) {
        return /(03|07|08|09|01[2|6|8|9])+([0-9]{8})\b/.test(number);
      },
      message: (Error) => "Không phải số điện thoại !",
    },
  },
  email: {
    type: String,
    required: true,
  },
  web: {
    type: String,
    required: true,
  },

  avatar: {
    type: String,
    default: "",
  },

  tokens: [{ token: { type: String, required: true } }],
});

const Company = mongoose.model("Company", CompanySchema); //đặt tên cho bản tạm thời

module.exports = Company; // cho phép các file khác có thể sử dụng file này (public file này lên)

File: client/lib/Models/company_model.dart


class Company{
  String name;
  String phone;
  String email;
  String web;

  Company({this.name, this.phone, this.email, this.web});

  Company.fromJson(Map<String, dynamic> json){
    name = json['name'];
    phone = json['phone'];
    email = json['email'];
    web = json['web'];
  }

  Map<String, dynamic> toJson(){
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['name'] = this.name;
    data['phone'] = this.phone;
    data['email'] = this.email;
    data['web'] = this.web;
  }
}



